### PR TITLE
Add preliminary style guide

### LINKED ISSUE #93: Define and audit doxgen style
I was a bad developer and started this project without clearly defined doxygen use guides. We should decide on a style and push it to the repo.

### PR DESCRIPTION
**Pull request by joshessman-llnl**
This is intended to be a very basic starting point for standardizing style and documentation.

The two initial amendments to Google's style guide are based on #90, but feel free to comment with any other amendments or additions you feel are useful for this first version.

### CODE REVIEW

File: src/docs/sphinx/style_guide.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+::
+
+    /** 
+     * Calculate du_dt = M^-1 (-Ku + f).
+     * This is all that is needed for explicit methods
+     * @param[in] u The state vector (input to the differentiation)
+     * @param[out] du The derivative of \p u with respect to time
+     * @see https://mfem.github.io/doxygen/html/classmfem_1_1TimeDependentOperator.html
+     */
```
Comment (samuelpmish) [inline src/docs/sphinx/style_guide.rst:45]: This will be a good change, I think most IDEs can pick up on these comments too

**Review by samuelpmish — APPROVED** [review]

File: src/docs/sphinx/index.rst [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 .. toctree::
    memory_checking
+   style_guide
```
Comment (white238) [inline src/docs/sphinx/index.rst]: Merge in develop so this gets added to the new hidden/at-the-bottom toctree

**Review by white238 — APPROVED** [review]
After fixing merge problem

**Review by jamiebramwell — APPROVED** [review]
Looks great modulo the merge @white238 mentioned!

**Comment by joshessman-llnl** [comment]
LGTM

**Pull request merged**